### PR TITLE
[OCPCLOUD-1706] Migrate test utils to actuator-pkg repository

### PR DESCRIPTION
**Pull request by RadekManak**
This PR moves the test and resourcebuilder package to https://github.com/openshift/cluster-api-actuator-pkg/pull/265

The test package is renamed to testutils.
The machineInfo resourcebuilder is left in this repository and is imported as localresourcebuilder.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,13 +21,16 @@ require (
 	k8s.io/klog/v2 v2.80.1
 	k8s.io/utils v0.0.0-20220823124924-e9cbc92d1a73
 	sigs.k8s.io/controller-runtime v0.13.0
-	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20220927155351-7399a3a595bf
+	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20221222054525-3c4deba74e5c
 	sigs.k8s.io/controller-tools v0.10.0
 )
 
+replace github.com/openshift/cluster-api-actuator-pkg => github.com/radekmanak/cluster-api-actuator-pkg v0.0.0-20230110123545-ca770eaa01a0
```
Comment (RadekManak) [inline go.mod:28]: /hold
Don't merge with this line.

**Comment by JoelSpeed** [comment]
In the test 
```
With a running controller with updated machines and the instance size is changed should perform a rolling update
/go/src/github.com/openshift/cluster-control-plane-machine-set-operator/pkg/controllers/controlplanemachineset/controller_test.go:1051
```
At step (Some steps run in parallel, this may be a red herring)
```
STEP: Checking the number of control plane machines never goes above 4 replicas @ 02/01/23 13:18:13.765
```
We are seeing the following race
```
==================
WARNING: DATA RACE
Read at 0x00c0001fe048 by goroutine 15994:
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitBlock()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:631 +0x56
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:553 +0x504
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).EmitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:540 +0x1e4
  github.com/onsi/ginkgo/v2/internal.(*Suite).handleSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:264 +0x2f4
  github.com/onsi/ginkgo/v2/internal.(*Suite).By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:285 +0x197
  github.com/onsi/ginkgo/v2.By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/core_dsl.go:524 +0x69
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.CheckRolloutForIndex()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/rollingupdate.go:37 +0x92
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.checkRolloutProgress()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/rollingupdate.go:85 +0x5d
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1.3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:95 +0x5d
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:85 +0xc4
Previous write at 0x00c0001fe048 by goroutine 15992:
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emit()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:623 +0x56
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitBlock()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:636 +0xf7
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:553 +0x504
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).EmitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:540 +0x1e4
  github.com/onsi/ginkgo/v2/internal.(*Suite).handleSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:264 +0x2f4
  github.com/onsi/ginkgo/v2/internal.(*Suite).By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:285 +0x197
  github.com/onsi/ginkgo/v2.By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/core_dsl.go:524 +0x69
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.CheckReplicasDoesNotExceedSurgeCapacity()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/rollingupdate.go:72 +0x64
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1.1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:87 +0x3e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:85 +0xc4
Goroutine 15994 (running) created at:
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:81 +0x12e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:94 +0x784
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:854 +0x114
Goroutine 15992 (running) created at:
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:81 +0x12e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:86 +0x5c4
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:854 +0x114
==================
```
And then again in step:
```
STEP: Waiting for the index 0 to be replaced @ 02/01/23 13:18:13.765
```
We are seeing this race
```
==================
WARNING: DATA RACE
Write at 0x00c0001fe058 by goroutine 15993:
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emit()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:624 +0x9a
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitBlock()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:634 +0xaf
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:553 +0x504
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).EmitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:540 +0x1e4
  github.com/onsi/ginkgo/v2/internal.(*Suite).handleSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:264 +0x2f4
  STEP: Checking that other indexes (not 0) do not have 2 replicas @ 02/01/23 13:18:13.766  github.com/onsi/ginkgo/v2/internal.(*Suite).By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:285 +0x197
  github.com/onsi/ginkgo/v2.By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/core_dsl.go:524 +0x69
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.WaitForControlPlaneMachineSetDesiredReplicas()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/controlplanemachineset.go:178 +0x178
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1.2()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:91 +0x84
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:85 +0xc4
Previous write at 0x00c0001fe058 by goroutine 15992:
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emit()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:624 +0x9a
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitBlock()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:636 +0xf7
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).emitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:553 +0x504
  github.com/onsi/ginkgo/v2/reporters.(*DefaultReporter).EmitSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/reporters/default_reporter.go:540 +0x1e4
  github.com/onsi/ginkgo/v2/internal.(*Suite).handleSpecEvent()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:264 +0x2f4
  github.com/onsi/ginkgo/v2/internal.(*Suite).By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:285 +0x197
  github.com/onsi/ginkgo/v2.By()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/core_dsl.go:524 +0x69
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.CheckReplicasDoesNotExceedSurgeCapacity()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/rollingupdate.go:72 +0x64
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1.1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:87 +0x3e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:85 +0xc4
Goroutine 15993 (running) created at:
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:81 +0x12e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:90 +0x6a4
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:854 +0x114
Goroutine 15992 (running) created at:
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework.Async()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/framework/utils.go:81 +0x12e
  github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers.ItShouldPerformARollingUpdate.func1()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/test/e2e/helpers/cases.go:86 +0x5c4
  github.com/onsi/ginkgo/v2/internal.extractBodyFunction.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/node.go:459 +0x30
  github.com/onsi/ginkgo/v2/internal.(*Suite).runNode.func3()
      /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/onsi/ginkgo/v2/internal/suite.go:854 +0x114
==================
```
I can see notes in there about this being a race within `framework.Async` so it could be an issue with how we are testing the various pieces of code in the waitgroup, will need to review the changes in ginkgo between 2.4 and 2.7 to identify if there's something obvious changed

**Comment by JoelSpeed** [comment]
By running just the single racing test locally (modify hack/test.sh):
```
#31 -  ${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ${TEST_PACKAGES}
#31 +  ${GINKGO} ${GINKGO_ARGS} ${GINKGO_EXTRA_ARGS} ./pkg/controllers/controlplanemachineset
```
And then focusing the test's parent context (pkg/controllers/controlplanemachineset/controller_test.go):
```
#1036 - Context("and the instance size is changed", func() {
#1036 + FContext("and the instance size is changed", func() {
```
And then using go modules replace statements to force the ginkgo version, I can see that the race isn't present in v2.4.0 (current version on main) but is present on v2.5.0.

So the issue should be introduced within this [diff](https://github.com/onsi/ginkgo/compare/v2.4.0...v2.5.0)

**Comment by damdo** [comment]
@RadekManak https://github.com/openshift/cluster-control-plane-machine-set-operator/pull/164 merged so if you want to rebase on top of `main` you should be good to go 👍

File: go.mod [hunk 2]
```diff
@@ -12,23 +12,26 @@ require (
...
 	k8s.io/klog/v2 v2.90.0
-	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448
+	k8s.io/utils v0.0.0-20230115233650-391b47cb4029
 	sigs.k8s.io/controller-runtime v0.14.4
-	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20230127152628-dd6bafd4572e
+	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20230131074648-f5014c077fc3
 	sigs.k8s.io/controller-tools v0.11.3
 )
 
+replace github.com/openshift/cluster-api-actuator-pkg/testutils => github.com/RadekManak/cluster-api-actuator-pkg/testutils v0.0.0-20230201125136-02f668983c91
```
Comment (JoelSpeed) [inline go.mod:33]: Just for tracking, this must be replaced before we merge

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.25
+ENVTEST_K9S_VERSION = 1.26
```
[the remark this answers is not in the document]
Comment (damdo) [reply Makefile]: 😎

**Review by damdo — COMMENTED** [review]
Thanks!
/lgtm

**Pull request merged**